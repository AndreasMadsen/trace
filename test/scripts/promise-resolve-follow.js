function f1() {
  return new Promise((resolve, reject) => {
    setTimeout(() => {
      reject(new Error('FAIL'));
    }, 0);
  }, 0);
}

function f2() {
  return new Promise((resolve, reject) => {
    f1().then(resolve, reject);
  });
}

function f3() {
  return Promise.resolve()
    .then(() => f2())
    .catch((err) => console.log(err));
}

f3();
