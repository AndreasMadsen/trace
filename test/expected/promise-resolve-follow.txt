Error: FAIL
    at Timeout.setTimeout [as _onTimeout] (/scripts/promise-resolve-follow.js:4:14)
    at ____________________:0:0
    at new Promise (<anonymous>)
    at f1 (/scripts/promise-resolve-follow.js:2:10)
    at Promise (/scripts/promise-resolve-follow.js:11:5)
    at ____________________:0:0
    at new Promise (<anonymous>)
    at f2 (/scripts/promise-resolve-follow.js:10:10)
    at Promise.resolve.then (/scripts/promise-resolve-follow.js:17:17)
    at <anonymous>
    at ____________________:0:0
    at Promise.catch (<anonymous>)
    at f3 (/scripts/promise-resolve-follow.js:18:11)
    at ____________________:0:0
    at Promise.then (<anonymous>)
    at f3 (/scripts/promise-resolve-follow.js:17:6)
    at ____________________:0:0
    at Function.resolve (<anonymous>)
    at f3 (/scripts/promise-resolve-follow.js:16:18)
    at Object.<anonymous> (/scripts/promise-resolve-follow.js:21:1)
